woocommerce/woocommerce-android
# Edit variation price

This PR refactors the product pricing screen by decoupling it from the product detail. There is a new `ProductPricingViewModel`, which takes over all price-editing logic from `ProductDetailViewModel`. The pricing screen is now independent and can be used for variation price editing.

The price screen UI should be the same. The same holds for product price editing behaviour. 

| Light | Dark |
|-|-|
| ![image](https://user-images.githubusercontent.com/1522856/89312251-55f82700-d677-11ea-8d87-a970e7682ca2.png) | ![image](https://user-images.githubusercontent.com/1522856/89312275-5f818f00-d677-11ea-8298-5d28b8df89ef.png) |

**To test:**

_Product details_
1. Select a simple product and go to the product details screen
2. Tap on the Price property and open the pricing screen
3. Smoke test price editing to make sure the functionality is correct
4. Make sure the Done button is visible only when there is a change
5. Check that the end sale date cannot be > start sale date
6. Check that the sale price cannot be > regular price
7. Check that the discard dialog appears if there is a change and back button is pressed
8. Make some changes, tap one the Done button
7. Verify that the changes are reflected in the product details

_Variation details_
1. Select a variable product
2. Tap on Variations property
3. Select a variation and open variation detail screen
4. Tap on the Price property
5. Notice the tax section is not visible
6. Smoke test price editing to make sure the functionality is correct
7. Make some changes, tap one the Done button
8. Verify that the changes are reflected in the product details

_Unit tests_
1. Run tests in `ProductPricingViewModelTest`
2. Verify all tests pass
